basic-data-types/percentage.py

```diff
@@ -0,0 +1,21 @@
+# Finding the percentage
+"""Task
+You have a record of N students. Each record contains the student's name, and their percent marks in Maths,
+Physics and Chemistry. The marks can be floating values. The user enters some integer N followed by the names and marks
+for N students. You are required to save the record in a dictionary data type. The user then enters a student's name.
+Output the average percentage marks obtained by that student, correct to two decimal places.
+"""
+import statistics
+
+
+if __name__ == '__main__':
+    n = int(input())
+    student_marks = {}
+    for _ in range(n):
+        name, *line = input().split()
+        scores = list(map(float, line))
+        student_marks[name] = scores
+    query_name = input()
+    for key in student_marks:
+        if query_name == key:
```

Comment:
you're looking over the whole dict which is the exactly opposite of it's intentions: fast key access, O(1) in average.
use insecure access in try/except block(we'll get there later):
```
try:
    marks = student_marks[query_name]
    print(...)
except KeyError:
    print("no such name")
```
or secure key access:
```
marks = student_marks.get(query_name)
if marks is not None:
    print(...)
else:
    print("no such name")
```

UPD. If the idea was to print **all** rows for the given `query_name` - here's the dict restriction - all keys are unique, so it will be only one row.